### PR TITLE
WT-7104 Redact user data from printlog output. (v4.2 backport)

### PR DESCRIPTION
**Pull request by boboeee**

### CODE REVIEW

**Comment by boboeee** [comment]
Currently have an email verification problem, but is ready to review

**Review by sueloverso — APPROVED** [review]

File: dist/log.py [hunk 1]
```diff
@@ -86,6 +86,19 @@ def printf_setup(f, i, nl_indent):
 def n_setup(f):
     return len(field_types[f[0]][4])
 
+# Check for an operation that has a file id type. Redact any user data
+# if the redact flag is set, but print operations for file id 0, known
+# to be the metadata.
+def check_redact(optype):
+    for f  in optype.fields:
```
[the remark this answers is not in the document]
Comment (boboeee) [reply dist/log.py]: done!

**Review by tetsuo-cpp — APPROVED** [review]
LGTM.
Nice work @boboeee.

**Comment by clarissecheah** [comment]
LGTM!

**Review by clarissecheah — APPROVED** [review]

**Pull request merged**